src/point_one/fusion_engine/messages/control.h

```diff
@@ -377,6 +377,75 @@ struct alignas(4) VersionInfoMessage : public MessagePayload {
   char fw_version_str[0];
 };
 
+/**
+ * @brief Device identifier information, (@ref MessageType::DEVICE_ID, version
+ *        1.0).
+ * @ingroup config_and_ctrl_messages
+ *
+ * This message contains ID data for each of the following, where available:
+ * - HW - A unique ROM identifier pulled from the device HW
+ * - User - A value set by the user to identify a device
+ * - Receiver - A unique ROM identifier pulled from the GNSS receiver
+ *
+ * The message payload specifies the length of each string (in bytes). It is
+ * followed by each of the listed ids consecutively. The values are _not_ null
+ * terminated and aren't necessarily strings. The interpretation of the strings
+ * can be distinguished based on the `device_type`.
+ *
+ * ```
+ * {MessageHeader, VersionInfoMessage, "HW ID", "User ID", "Receiver ID"}
+ * ```
+ */
+struct alignas(4) DeviceIDMessage : public MessagePayload {
+  static constexpr MessageType MESSAGE_TYPE = MessageType::DEVICE_ID;
+  static constexpr uint8_t MESSAGE_VERSION = 0;
+
+  /** The type of device this message originated from. */
+  enum class DeviceType : uint8_t {
+    /**
+     * Unable to map device to a defined entry.
+     */
+    UNKNOWN = 0,
+    /** Device is a PointOne Atlas. */
+    ATLAS = 1,
+    /** Device is a Quectel LG69T-AM system. */
+    LG69T_AM = 2,
+    /** Device is a Quectel LG69T-AP system. */
+    LG69T_AP = 3,
+    /** Device is a Quectel LG69T-AH system. */
+    LG69T_AH = 4,
+  };
```

Comment:
Might be nice to add to string and << for this enum like the others. Also may want to define it outside this message struct in case we want to use it somewhere else in the future.